lib/modem.js

```diff
@@ -529,13 +529,11 @@ Modem.prototype.followProgress = function (streama, onFinished, onProgress) {
 Modem.prototype.buildQuerystring = function (opts) {
   var clone = {};
 
-  // serialize map and array values as JSON strings, else querystring truncates.
-  // 't' and 'extrahosts' can be arrays but need special treatment so that they're
-  // passed as multiple qs parameters instead of JSON values.
   Object.keys(opts).map(function (key, i) {
-    if (opts[key]
-      && typeof opts[key] === 'object'
-      && !['t', 'extrahosts'].includes(key)
+    if (
+      opts[key] &&
+      typeof opts[key] === 'object' &&
+      !Array.isArray(opts[key])
```

Comment:
@Clovel @apocas  This change is quite aggressive, it goes for using query serialization for only `t` and `extrahosts` to using it for any `Array` option. 

This doesn't seem to be universally true. At least I can see this breaks the `cachefrom` on Image.build that expects a JSON serialized array and otherwise it errors with 

```
(HTTP code 400) unexpected - error reading cache-from: invalid character 'a' looking for beginning of value
```  

I confirmed the issue does not happen with docker-modem v5.0.3, prior to this change.

Reply:
It would seem that the serialization needs to be more standard indeed. Not all routes behave the same (or at least I haven't found a common denominator). 

It does fix the usage of ENV variables and other parameters of the `/images/create` route though, which was my issue.

Also unit tests pass, so a case seems to be missing for your issue.

Reply:
In the Moby code, `t` and `extrahosts` are read using [Request.Form](https://pkg.go.dev/net/http#Request.Form) (see [lines 55-56](https://github.com/moby/moby/blob/a2a6bf51fae014453a8f60058270a3a052e0e529/api/server/router/build/build_routes.go#L55-L56)), while `cachefrom` uses [Request.FormValue](https://pkg.go.dev/net/http#Request.FormValue) (see [line 143](https://github.com/moby/moby/blob/a2a6bf51fae014453a8f60058270a3a052e0e529/api/server/router/build/build_routes.go#L143)). I don't know enough about Go to understand why the data is interpreted differently, but that could give us a clue.

Reply:
Yeah, the docker API is not very consistent with how it accepts array parameters. 

- `cachefrom`  in  [ImageBuild](https://docs.docker.com/reference/api/engine/version/v1.48/#tag/Image/operation/ImageBuild) expects a stringified "JSON array of images"
- `changes` in [ImageCreate](https://docs.docker.com/reference/api/engine/version/v1.48/#tag/Image/operation/ImageCreate) expects an array of strings.

I'll open up an issue and possibly a PR